lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -104,12 +103,12 @@
     private final List<SessionPropertyAccessControlRule> sessionPropertyRules;
     private final Set<AnySchemaPermissionsRule> anySchemaPermissionsRules;
 
-    public FileBasedAccessControl(CatalogName catalogName, File configFile)
+    public FileBasedAccessControl(CatalogName catalogName, Supplier<AccessControlRules> rulesProvider)
```

Comment:
Can we rename it as `JsonAccessControl` or something ? 

cc: @kokosing Any good names we can use here ?

Reply:
You mean rename the file/class? We have a few classes starting with FileBasedAccessControl? Do you want to rename all of them?

Reply:
refactored - renamed to RulesBasedAccessControl